server/mneme_api_response.rb

```diff
@@ -80,6 +81,49 @@ def extractAssignment(assignment)
     assign
   end
 
+  ############ Filter assignments
+  # The filter will take a single assignment and return it if it is acceptable and return nil
+  # if the assignment should not be used.
+
+  # The checks require date comparisons. To make testing easier the actual filter is implemented as
+  # a class method which requires the time as an argument.  The following instance method defaults
+  # the time to the current time to now.  'now' is a method that can be overridden by tests.
+
+  # This is an instance method to bind the time used for filtering to the current time.
+  def filter(assignment)
+    result = self.class.filter_two(assignment, Time.now().to_i)
+    logger.debug "#{self.class.to_s}:#{__method__}: #{__LINE__}: skipping assignment: [#{assignment.inspect}]" unless result
+    return result
+  end
+
+  # This is a class method for easy testing.  The current time must be passed in
+  def self.filter_two(assignment, now)
```

Comment:
The name is "filter" to describe the stage in processing.  Other providers use the same workflow and it is easier to understand (and generalize) processing if the workflow steps have the same name.  I don't like "filter_two" either.  I'll see if your name is better.

Time is passed in to make automated testing easier.  Testing against a real "now" is complicated and tests should be simple.  

I'll see if using an optional value where the default value is a method call works on a static method.